src/sdam/server_description.ts

```diff
@@ -105,6 +105,7 @@ export class ServerDescription {
     if (options?.topologyVersion) {
       this.topologyVersion = options.topologyVersion;
     } else if (hello?.topologyVersion) {
+      // TODO(NODE-2674): Preserve int64 sent from MongoDB
```

Comment:
while you're in this file, could you also fix the null access issue in `equals`? we access `other.topologyVersion` in L185, but in L194 we check that `other != null`